src/components/SocialMediaLinks/SocialMediaLinks.tsx

```diff
@@ -0,0 +1,195 @@
+      console.error(
+        "NYPL Reservoir SocialMediaLinks: " +
+          myPlatform.type +
+          " is either not a supported platform or is misspelled.\n" +
+          "Supported platforms include: " +
+          socialMediaLinkTypeArray.join(", ")
+      );
+      return;
+    }
+
+    // Get the dataset for this platform.
+    let thisPlatformArray = socialMediaDataMap.filter(
+      (socialMediaSlice) => socialMediaSlice.type === myPlatform.type
+    );
+
+    // The filter function returns an array of objects, so let's just get the objects out of the array.
+    let thisPlatformData = thisPlatformArray[0];
+
+    // If a url value exists in myPlatform use it, else use the default.
+    let newUrl = myPlatform.url ?? thisPlatformData.url;
+
+    // If a labelText value exists in myPlatform use it, else use the default.
+    let newLabelText = myPlatform.labelText ?? thisPlatformData.labelText;
+
+    let thisObj = {
+      type: thisPlatformData.type,
+      iconName: thisPlatformData.iconName,
+      labelText: newLabelText,
+      url: newUrl,
+    };
+    allData.push(thisObj);
+  }); // end foreach
+
+  return allData;
+}
+
+/**
+ * The SocialMediaLinks component renders a list of links for accessing social media sites.
+ */
+export const SocialMediaLinks = chakra(
+  forwardRef<HTMLDivElement, SocialMediaLinksProps>((props) => {
+    const {
+      borders = "none",
+      color = "textDefault",
+      className,
+      id,
+      linksData,
+      layout = "row",
+      showLabels = false,
+      size = "medium",
+      ...rest
+    } = props;
+
+    // Turns out you can pass whatever props you want to this thing in order to do logic in the theme.
+    const styles = useMultiStyleConfig("SocialMediaLinks", {
+      variant: borders,
+      size: size,
+      color: color,
+      layout: layout,
+    });
+
+    if (showLabels && borders === "circular") {
+      console.error(
+        "NYPL Reservoir SocialMediaLinks: `showLabels` is set to true, but labels are disallowed when also using borders='circular'."
+      );
+      return (
+        <p>
+          <Icon name={"alertWarningFilled"} size={"xlarge"} color={"red"} />
+          Error: See Console Log
+        </p>
+      );
+    }
+
+    // If linksData has values, use them, else use the entire list of platforms.
+    const socialMediaDataArray = linksData
+      ? getLinksData(linksData)
+      : socialMediaDataMap;
+
+    // Loop through the platform data array and build an array of links.
+    const thisLinksData = [];
+    socialMediaDataArray.forEach((myPlatform) => {
+      // The size prop for the Icon component does not exactly match the convention for SocialMediaLinks size prop.
+      // So let's set the correct Icon size.
+      let iconSize: IconSizes = "medium";
+      switch (size) {
+        case "medium":
+          iconSize = "large";
+          break;
+        case "large":
+          iconSize = "xlarge";
```

Comment:
add a `break` here as well.

Reply:
OK. But it works fine w/o it.

Reply:
Yes, but better to stick with JS convention and expectations when reading the code.